src/GroupManager.php

```diff
@@ -77,12 +77,12 @@ public function getGroupsForEntityType($entity_type_id) {
    *
    * @return array
    */
-  public function getAllGroups() {
+  public function GetAllGroupBundles($entity_type = NULL) {
     if (!isset($this->groupMap)) {
       $this->refreshGroupMap();
     }
 
-    return $this->groupMap;
+    return !empty($this->groupMap[$entity_type]) ? $this->groupMap[$entity_type] : $this->groupMap;
```

Comment:
Kind of :) If you passing `entity_type_not_a_group` there would be nothing in the group map, but you will be passing in an entity type. So an empty array should be returned in that case, not all data? If $entity_type is explicitly NULL then all data should be returned.